### PR TITLE
port Firestore SnapshotVersion in C++

### PR DESCRIPTION
**Pull request by zxu123**
### Discussion
part of the C++ migration

### Testing
unit test

### API Changes
n/a

### CODE REVIEW

File: Firestore/core/src/firebase/firestore/core/CMakeLists.txt [hunk 1]
```diff
@@ -17,6 +17,8 @@ cc_library(
   SOURCES
     database_info.cc
     database_info.h
+    snapshot_version.cc
```
Comment (wilhuff) [inline Firestore/core/src/firebase/firestore/core/CMakeLists.txt]: SnapshotVersion is in the model package on Android and we should have fixed this on iOS but didn't. Can you please move this `model` to line up with Android?

File: Firestore/core/src/firebase/firestore/core/snapshot_version.h [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+ public:
+  explicit SnapshotVersion(
+      const firebase::firestore::model::Timestamp& timestamp);
+
+  const firebase::firestore::model::Timestamp& timestamp() const {
+    return timestamp_;
+  }
+
+  /** Creates a new version that is smaller than all other versions. */
+  static const SnapshotVersion& NoVersion();
```
Comment (wilhuff) [inline Firestore/core/src/firebase/firestore/core/snapshot_version.h]: `NoVersion` is the Objective-C name for this property. Android calls this `None` which is a better name.

In general for names of these kinds of things we want to end up closer to Android/JS than to the old Objective-C names.

File: Firestore/core/src/firebase/firestore/core/CMakeLists.txt (hunk 1 above)
Comment (zxu123) [reply Firestore/core/src/firebase/firestore/core/CMakeLists.txt]: done

File: Firestore/core/src/firebase/firestore/core/snapshot_version.h (hunk 2 above)
Comment (zxu123) [reply Firestore/core/src/firebase/firestore/core/snapshot_version.h]: done

**Pull request merged**